### PR TITLE
chore: add unwaited futures to analysis options

### PR DESCRIPTION
**Pull request by richardshiue**
### Feature Preview

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/mobile/presentation/setting/font/font_setting.dart [hunk 1]
```diff
@@ -38,7 +40,9 @@ class FontSetting extends StatelessWidget {
         if (newFont != null && newFont != selectedFont) {
           if (context.mounted) {
             context.read<AppearanceSettingsCubit>().setFontFamily(newFont);
-            context.read<DocumentAppearanceCubit>().syncFontFamily(newFont);
+            unawaited(
+              context.read<DocumentAppearanceCubit>().syncFontFamily(newFont),
+            );
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/setting/font/font_setting.dart:45]: I think the cleanup is good, but the lint forces us to use this `unawaited`, which I don't particularly like 😂

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/lib/mobile/presentation/setting/font/font_setting.dart (hunk 1 above)
Comment (richardshiue) [reply frontend/appflowy_flutter/lib/mobile/presentation/setting/font/font_setting.dart:45]: Yes, that's the downside.

Comment (appflowy) [reply frontend/appflowy_flutter/lib/mobile/presentation/setting/font/font_setting.dart:45]: HAHA. I agreed with @Xazin . However, there may be a few rare instances where `unawaited` is necessary

**Comment by richardshiue** [comment]
Seems like some futures aren't getting completed.

**Pull request merged**